src/Client.php

```diff
@@ -474,10 +475,41 @@ protected function post($endpoint, array $body, $path = null, $idempotencyKey =
 
             return $responseArray[$endpoint];
         } catch (BadResponseException $e) {
+            if ($this->isIdempotentCreationConflict($e)) {
+                throw IdempotentCreationConflictException::fromBadResponseException($e);
+            }
+
             throw ApiException::fromBadResponseException($e);
         }
     }
 
+    /**
+     * @param BadResponseException $e
+     * @return bool
+     */
+    private function isIdempotentCreationConflict(BadResponseException $e)
```

Comment:
It will be nice when the code moves away from the ambiguous `$e` to a more contextual name like `$exception` 😉